skt/reporter.py

```diff
@@ -234,6 +234,24 @@ def __init__(self, cfg):
         # multireporting
         self.multi_job_ids = []
 
+    def get_config_local(self):
+        """Retrieve a local config file from the disk."""
+        config_file = "{}/.config".format('workdir')
+        if os.path.isfile(config_file):
+            # The config file
+            with open(config_file, "r") as fileh:
+                return fileh.read()
+
+        return None
+
+    def get_config_remote(self):
+        """Retrieve a config file from a remote location."""
```

Comment:
Same here, it needs to say "the published config file", and needs to either define publishing, or to refer to another place where it's done. Major part of why it's so difficult to describe these things here is that we keep passing the "cfg" object (essentially a bunch of global variables) around, instead of just what we need and no more. Each place has to keep track of what the rest of the program does, and know about (and refer to) all the events and entities it can possibly interact with.

Reply:
Fixed.